### PR TITLE
Update to npm module to fix issue #1375 

### PR DESCRIPTION
**Pull request by gunzy83**
This fixes issue #1375 by checking for a missing package and issuing an **install()** instead of an **update()**.

### CODE REVIEW

**Comment by gunzy83** [comment]
Appears tests are failing for an unrelated reason on multiple PRs.

File: packaging/language/npm.py [hunk 1]
```diff
@@ -253,7 +253,10 @@ def main():
     elif state == 'latest':
         installed, missing = npm.list()
         outdated = npm.list_outdated()
-        if len(missing) or len(outdated):
+        if len(missing):
+            changed = True
+            npm.install() 
+        elif len(outdated):
             changed = True
```
Comment (nitzmahone) [inline packaging/language/npm.py:259]: Should this actually just be a separate if (not an elif)? If both lists are non-empty, you'll only do the missing ones...

Comment (gunzy83) [inline packaging/language/npm.py:259]: Correct. That was one cause I failed to test.

I have updated the PR.

Comment (RichVel) [inline packaging/language/npm.py:259]: Any chance this PR could be merged?  Issue #1375 is still open, and a related issue is no. 2 result on Google for "ansible npm module", so it's probably happening a lot.

**Comment by shane-walker** [comment]
I'm also having this problem, it is making me do annoying things just to make things work...
@chrishoffman is there any way we can get this merged?

``` yaml
- name: install NPM v3 (this is a workaround)
  npm: name=npm version=3 state=present global=yes production=yes

- name: update NPM to latest release of v3
  npm: name=npm version=3 state=latest global=yes production=yes
```

**Comment by chrishoffman** [comment]
I haven't really done any node development in over 3 years.  I think it makes sense that someone else begin to maintain this module.  I'm not sure how to make that happen but I think it would help with the ongoing support and maintenance of this module.

**Comment by shane-walker** [comment]
@gregdek @resmo What's the process for adopting a module? It looks like @chrishoffman is interested in handing over the reigns for this one. I would be willing to volunteer for the NPM module if necessary.

**Comment by conorsch** [comment]
Thanks for stepping up, @shane-walker. I was just stepping through the module code and found my way to this issue. @chrishoffman Thanks for your hard work, and also for your candor in trying to find a new maintainer!

**Comment by gregdek** [comment]
@shane-walker thanks for volunteering. I'll go ahead and set up up as the maintainer of record for this one.

**Comment by gregdek** [comment]
(And big thanks to @chrishoffman for care and feeding of this module for so long.)

**Comment by charlienewey** [comment]
Amazing. I was just coming here to write this PR myself. Thanks, @shane-walker!

File: packaging/language/npm.py (hunk 1 above)
Comment (gunzy83) [reply packaging/language/npm.py:259]: Apparently my push didn't push. Things should be up to date now based on the comment above.

**Comment by resmo** [comment]
Thanks for this PR @gunzy83! 

Unfortunately I merged #3020 by @shane-walker addressing the same thing and didn't realize your PR is identical. I feel like I should have merged your PR, sorry for that.

Closing this PR by #3020

**Pull request closed without merging**